### PR TITLE
More changes for billing / bridge-gui

### PR DESCRIPTION
**Pull request by bryanchriswhite**

### CODE REVIEW

File: lib/models/payment-processor-adapters/stripe.js [hunk 1]
```diff
@@ -60,20 +66,22 @@ const stripeAdapter = {
...
-            }
-
-            if (!confirmation.deleted) {
-              return reject(new Error('stripe couldn\'t delete the customer'));
-            }
-
-            return resolve(null);
-          });
-    });
+    throw new Error("you shouldn'n be here");
```
Comment (braydonf) [inline lib/models/payment-processor-adapters/stripe.js]: Can we clean this up? e.g. `throw new Error('Not implemented')`, and the existing code will be in git history (no need to comment section of code).

**Comment by braydonf** [comment]
There are several cases where it looks like an error should be sent with `reject`, but instead it logs or ignores completely. Using specific error types may help with passing along the error, and being able to handle it in the layer above.

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "dotenv": "^2.0.0",
     "elliptic": "^6.3.2",
     "hat": "0.0.3",
+    "lodash": "^4.17.4",
```
Comment (braydonf) [inline package.json]: `lodash` actually only looks to be used in a test. Add to devDependency?

**Comment by bryanchriswhite** [comment]
@braydonf thanks for the review :D

You're correct that there are probably several places where there is an error and `reject` isn't called (or a similar scenario).

I'm more than happy to go through specific examples if you recall where any are but in general I believe that most if not all of those are likely places where our api is calling out to stripe and we're handling error cases in the callback rather than throwing errors to be handled elsewhere. I'm open to discussing a different approach if you like though.

**Comment by barbaraliau** [comment]
Merging (▀̿Ĺ̯▀̿ ̿)

**Pull request merged**